src/frontend/components/VideoPlayer/VideoPlayer.tsx

```diff
@@ -1,20 +1,23 @@
-import Plyr from 'plyr';
 import 'plyr/dist/plyr.css';
 import * as React from 'react';
 import { Redirect } from 'react-router';
 import shaka from 'shaka-player';
 
+import { createPlayer } from '../../Player/createPlayer';
 import { Video, videoSize } from '../../types/tracks';
+import { VideoPlayerInterface } from '../../types/VideoPlayerInterface';
 import { Maybe, Nullable } from '../../utils/types';
 import { ROUTE as ERROR_ROUTE } from '../ErrorComponent/ErrorComponent';
 import './VideoPlayer.css'; // Improve some plyr styles
 
 export interface VideoPlayerProps {
+  jwt: string;
   video: Nullable<Video>;
+  createPlayer: createPlayer;
 }
 
 interface VideoPlayerState {
-  player: Maybe<Plyr>;
+  player: Maybe<VideoPlayerInterface>;
```

Comment:
I don't think we can have this component be player-agnostic.

If we end up supporting several players, each one would probably need to have its own component and then an overarching one to launch the correct player (and maybe allow user choice?) and pass the relevant data.

Reply:
We will only defined the method we need in our component. For now we only need the destroy method. If this method does not exists in a future player we will create a decorator that implement it. IMO we need this interface

Reply:
What I meant is that IMO `VideoPlayer.tsx` is really `VideoPlayerPlyr.tsx` and well need a `VideoPlayerWhatever.tsx` when we implement another player.

That is, we'll probably have to remove the `createPlayer` method as soon as we implement another player. Anyway, I guess we can leave it this way for now as there is only one player